### PR TITLE
fix(general): update checks_metadata structure

### PR DESCRIPTION
**Pull request by arielkru**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
check_meatada.json should be a dict of resourceIds as keys, with each of the resourceIds containing a dict with check_ids as keys and the metadata as values.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/bridgecrew/wrapper.py [hunk 1]
```diff
@@ -37,9 +38,14 @@ def _put_json_object(s3_client: BaseClient, json_obj: Any, bucket: str, object_p
 
 
 def _extract_checks_metadata(report: Report, full_repo_object_key: str) -> dict[str, dict[str, Any]]:
-    return {check.check_id: dict({k: getattr(check, k, "") for k in check_metadata_keys},
-                                 **{'file_object_path': full_repo_object_key + check.file_path}) for check in
-            list(itertools.chain(report.passed_checks, report.failed_checks, report.skipped_checks))}
+    metadata = defaultdict(dict)
+    for check in list(itertools.chain(report.passed_checks, report.failed_checks, report.skipped_checks)):
```
Comment (marynaKK) [inline checkov/common/bridgecrew/wrapper.py:42]: no need for list - the itertoolss.chain is enough :)

File: checkov/common/bridgecrew/wrapper.py [hunk 2]
```diff
@@ -37,9 +38,14 @@ def _put_json_object(s3_client: BaseClient, json_obj: Any, bucket: str, object_p
...
+        metadata_key = f'{check.file_path}:{check.resource}'
+        check_meta = dict({k: getattr(check, k, "") for k in check_metadata_keys},
```
Comment (marynaKK) [inline checkov/common/bridgecrew/wrapper.py:44]: dict() can be replaced with {} : https://katrin-affolter.ch/Programming/Python_dictionaries
much more faster and better :)

**Review by marynaKK — COMMENTED** [review]
had 2 comments :) 🍒 
Also, a Q as I do not fully understand - are the resource_id the same as the one we are changing in places like gha, azure pipelines, IR? if they are - they are different from the id that is used here. Is this an issue?

**Comment by arielkru** [comment]
> had 2 comments :) 🍒 Also, a Q as I do not fully understand - are the resource_id the same as the one we are changing in places like gha, azure pipelines, IR? if they are - they are different from the id that is used here. Is this an issue?

@marynaKK good q, it shouldn't be an issue because the resourceId here doesnt change the resourceIds in the reports, it just fits the way we expect resource keys to appear in the checks_metadata.json file.
So for example an azure_pipelines resource is saved under the key: `/azure-pipelines.yaml:jobs[0](FailTag)` and that's ok

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**